nobrainer/metrics.py

```diff
@@ -63,10 +54,14 @@ def generalized_dice(y_true, y_pred, axis=(1, 2, 3)):
         raise ValueError("y_true and y_pred must be at least rank 2.")
 
     epsilon = tf.keras.backend.epsilon()
-    w = tf.math.reciprocal(tf.square(tf.reduce_sum(y_true, axis=axis)) + epsilon)
+
+    w = tf.math.reciprocal(tf.square(tf.reduce_sum(y_true, axis=axis)))
+    w = tf.where(tf.math.is_finite(w), w, epsilon)
```

Comment:
I'm wondering why we should get rid of the small epsilon in favor of checking where the value is finite. In papers I've read, implementations typically use a small epsilon to avoid division by zero. Is this proposed solution more stable?

Reply:
Yes, this is more stable. For classes not present, 0/ep is not favorable.